changelog.d/11226.bugfix

```diff
@@ -0,0 +1 @@
+Fix a bug in unit test `test_block_room_and_not_purge`.
```

Comment:
I think this should be `11226.misc` because it's not a user-visible bug (rather, a mistake and hole in our test coverage).

Otherwise I'm happy with this. I'll open a separate PR to make the api return 400 if you ask for both block and purge to be False.